book/spiders/books.py

```diff
@@ -0,0 +1,43 @@
+import scrapy
+from scrapy.http.response import Response
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs: dict) -> None:
+        for book in response.css(".product_pod"):
+            book_page = response.urljoin(
+                book.css(".product_pod h3 a::attr(href)").get()
+            )
+            yield scrapy.Request(book_page, callback=self.parse_single_book)
+
+        next_page = response.css("li.next a::attr(href)").get()
+
+        if next_page is not None:
+            yield response.follow(next_page, callback=self.parse)
+
+    @staticmethod
+    def parse_single_book(book_page: Response) -> dict[str, str]:
+        detail_info = {
```

Comment:
We can use "yield" here instead of yielding this dict at the end of the function